File: elixir/bob/lib/bob.ex
defmodule Bob do
  @spec hey(String.t()) :: String.t()

  def hey(""), do: "Fine. Be that way!"

  def hey(input) do
    cond do
      "" == String.trim(input) ->
        "Fine. Be that way!"

      "?" == String.last(String.trim(input)) and input == String.upcase(String.trim(input)) and
          String.downcase(input) != input ->
        "Calm down, I know what I'm doing!"

      "?" == String.last(String.trim(input)) ->
        "Sure."

      input == String.upcase(input) and String.downcase(input) != input ->
        "Whoa, chill out!"

      true ->
        "Whatever."
    end
  end
end

File: java/bob/src/main/java/Bob.java
/*

Since this exercise has a difficulty of > 4 it doesn't come
with any starter implementation.
This is so that you get to practice creating classes and methods
which is an important part of programming in Java.

Please remove this comment when submitting your solution.

*/

public class Bob {
    public Bob() {
    }

    public String hey(String input) {
        if (input.trim().isEmpty()) {
            return "Fine. Be that way!";
        }

        if (input.equals(input.toUpperCase()) && input.matches(".*[a-zA-Z].*") && input.endsWith("?")) {
            return "Calm down, I know what I'm doing!";
        }
        if (input.equals(input.toUpperCase()) && input.matches(".*[a-zA-Z].*")) {
            return "Whoa, chill out!";
        }

        if (input.trim().endsWith("?")) {
            return "Sure.";
        }
        return "Whatever.";
    }
}
